### DISCUSSION TITLE
Log Result Reason Documentation and explanation

### DISCUSSION THREAD

**Discussion by ar3thien** [AdguardTeam/AdGuardHome#6807] [General]
Hello,
I'm parsing the logs in grafana/loki. I'm currently looking for an official documentation that explains different "Reason Code" found in the result of every query. Example:

{
"T":"2024-03-06T13:27:48.739753825Z",
"QH":"xxxxxxxxxx",
"QT":"A",
"QC":"IN",
"CP":"",
"Answer":"xxxxxxxxxx",
"IP":"192.168.x.x",
"Result":{
"CanonName":"xxxxxxx",
"IPList":[
"192.168.x.x"
],
**"Reason":9**
},
"Elapsed":43277
}

I've multiple Reason Codes in the logs. So far I'm having code 9 , 3 , and 7. any one knows what those mean? and in case there is an official doc about them?

**Comment by hagezi** [comment]
@ar3thien https://github.com/AdguardTeam/AdGuardHome/blob/bcd1430680f4d8bccb8d262edbdb48b05b918326/internal/filtering/filtering.go#L285

```
0 = NotFilteredNotFound
...
3 = FilteredBlockList
...
7 = FilteredSafeSearch
...
9 = Rewritten
...
```

**Reply by ar3thien** [reply]
Thank you so much @hagezi . So basically Reason Id seen in the logs, is the index of every reason string in the following [array](https://github.com/AdguardTeam/AdGuardHome/blob/bcd1430680f4d8bccb8d262edbdb48b05b918326/internal/filtering/filtering.go#L332). Am I right?

**Reply by hagezi** [reply]
In my opinion, that's how it should be, you can check it with filtered domains.

**Comment by JBHorne2** [comment]
**Type	Condition	Display	Color**
Special	Empty	Not filtered	green
Value	1	Allowed by allowlist	green
Value	2	Filtering error	orange
Value	3	Blocklist rule	red
Value	4	Safe browsing	red
Value	5	Parental control	red
Value	6	Invalid request	orange
Value	7	Safe search rewrite	blue
Value	8	Blocked service	red
Value	9	Rewrite (legacy rule)	blue
Value	10	Rewrite (/etc/hosts)	blue
Value	11	Rewrite ($dnsrewrite)	blue
